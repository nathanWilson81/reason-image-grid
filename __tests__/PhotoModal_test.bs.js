// Generated by BUCKLESCRIPT VERSION 5.0.6, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("@glennsl/bs-jest/src/jest.js");
var React = require("react");
var JestDom = require("bs-jest-dom/src/JestDom.bs.js");
var ReactTestingLibrary = require("bs-react-testing-library/src/ReactTestingLibrary.bs.js");
var PhotoModal$ReactHooksTemplate = require("../src/PhotoModal.bs.js");

Jest.describe("PhotoModal", (function (param) {
        var photo = /* record */[
          /* albumId */1,
          /* id */1,
          /* title */"This is the title",
          /* url */"foo1",
          /* thumbnailUrl */"foo2"
        ];
        var photos = /* :: */[
          photo,
          /* [] */0
        ];
        var onInputSubmit = (jest.fn());
        var setModalClosed = (jest.fn());
        Jest.test("renders the modal", (function (param) {
                var eta = React.createElement(PhotoModal$ReactHooksTemplate.make, {
                      photo: photo,
                      photos: photos,
                      setModalClosed: setModalClosed,
                      onInputSubmit: onInputSubmit
                    });
                return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](ReactTestingLibrary.render(undefined, undefined, eta).container));
              }));
        return Jest.test("title shows", (function (param) {
                      var partial_arg = /* `Str */[
                        4153489,
                        "This is the title"
                      ];
                      var eta = React.createElement(PhotoModal$ReactHooksTemplate.make, {
                            photo: photo,
                            photos: photos,
                            setModalClosed: setModalClosed,
                            onInputSubmit: onInputSubmit
                          });
                      return JestDom.toBeInTheDocument(Jest.Expect[/* expect */0]((function (eta) {
                                          var param = undefined;
                                          var param$1 = eta;
                                          return ReactTestingLibrary.getByText(partial_arg, param, param$1);
                                        })(ReactTestingLibrary.render(undefined, undefined, eta))));
                    }));
      }));

/*  Not a pure module */
