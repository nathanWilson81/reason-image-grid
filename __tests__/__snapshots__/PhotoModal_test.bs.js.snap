// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoModal renders the modal 1`] = `
<div>
  <div
    class="css-r88ox7"
  >
    <div>
      <h3>
        This is the title
      </h3>
      <img
        src="foo1"
      />
      <div
        class="css-k008qs"
      >
        <input
          value=""
        />
        <button>
          Submit
        </button>
        <button>
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
